py/desisim/lya_spectra.py

```diff
@@ -46,40 +48,45 @@ def get_spectra(infile, first=0, nqso=None, seed=None):
     rand = sp.random.RandomState(seed)
     seed = rand.randint(2**32, size=nqso)
 
-    input_meta = empty_metatable(objtype='QSO', nmodel=nqso)
+    input_meta = empty_metatable(objtype='QSO', nmodel=1)
 
     filter_name = 'sdss2010-g'
 
     normfilt = filters.load_filters(filter_name)
     qso = QSO(normfilter=filter_name)
 
-    input_meta["REDSHIFT"]=zqso
-    input_meta["MAG"]=mag_g
-    input_meta["SEED"]=seed
-
-    flux, wave, meta = qso.make_templates(input_meta=input_meta)
-    
-    # Add RA,DEC to output meta
-    meta['RA'] = ra
-    meta['DEC'] = dec
-
+    flux = np.zeros([nqso, len(qso.wave)])
+    meta = None
     for i,head in enumerate(h[first+1:first+1+nqso]):
+        f, wave, meta_qso = qso.make_templates(nmodel=1,
+                                               redshift=np.array([zqso[i]]), mag=np.array([mag_g[i]]), seed=seed[i])
+
```

Comment:
We can generate N>>1 QSO spectra, but it's superslow. Having the loop here is the key to the 10x speedup.